CRM/MembershipExtras/Service/MembershipInstalmentsSchedule.php

```diff
@@ -137,7 +145,16 @@ private function getMembershipStartDate(DateTime $startDate = NULL, DateTime $en
    */
   private function calculateInstalmentAmount() {
     if ($this->membershipTypes[0]->period_type == 'fixed') {
-      $instalmentAmount = new InstalmentAmount(new FixedPeriodCalculator($this->membershipTypes));
+      if ($this->schedule == self::MONTHLY) {
```

Comment:
Maybe this could be cleaned up by refactoring to a new method.

Eg.
```php
private function calculateInstalmentAmount() {
  $instalmentAmount = $this->getInstalmentAmountObject();
  $instalmentAmount->getCalculator()->calculate();
  ...
}

private function getInstalmentAmountObject() {
  if ($this->membershipTypes[0]->period_type == 'fixed') {
    if ($this->schedule == self::MONTHLY) {
        $fixedPeriodTypeMonthlyCalculator = new FixedPeriodTypeMonthlyCalculator($this->membershipTypes);
        $fixedPeriodTypeMonthlyCalculator->setStartDate($this->startDate);
        $instalmentAmount = new InstalmentAmount($fixedPeriodTypeMonthlyCalculator);
    }
    else {
      ...
    }
  }
  
  return $instalmentAmount;
}
```

Reply:
@MiyaNoctem  nice imrovment!